dkms.in

```diff
@@ -1040,22 +1107,52 @@ prepare_build()
             $"Check $build_dir for more information."
     done
 
-    if [[ -e "${kernel_config}" ]]; then
-        if grep -q 'CONFIG_CC_IS_CLANG=y' "${kernel_config}"; then
-            local cc=clang
-            if command -v "$cc" >/dev/null; then
-                export CC="$cc"
-                export KERNEL_CC="$cc"
+    # Detect CC and LD used by kernel
+    if ! grep -qE '(^|\s)CC=' <<< "${make_command}"; then
+        # No CC specified within MAKE in dkms.conf
+        if [[ -e "${kernel_config}" ]]; then
+            if grep -q '^CONFIG_CC_IS_GCC=y' "${kernel_config}"; then
+                # Since kernel 4.18
+                KERNEL_CC="$(detect_toolchain_from_kconfig GCC gcc)"
+            elif grep -q '^CONFIG_CC_IS_CLANG=y' "${kernel_config}"; then
+                # Since kernel 4.18
+                KERNEL_CC="$(detect_toolchain_from_kconfig CLANG clang)"
+            elif [[ -f "$kernel_source_dir/.kernelvariables" ]]; then
+                # Debian specific
+                KERNEL_CC="$(echo -e "show-%:\n\t@echo \$(\$*)\ninclude $kernel_source_dir/.kernelvariables" | make -f - show-CC)"
```

Comment:
In my testing, `detect_toolchain_from_kconfig` works correctly and this `kernelvariables` workaround could be dropped. Not a big deal if you want to keep it, but in practise it's untested and never used.